### PR TITLE
OIDC federated idp init logout Integration tests

### PR DESCRIPTION
**Pull request by ThaminduR**
Add integration tests for OIDC federated identity provider initiated logout.

Related to - https://github.com/wso2/product-is/issues/9254

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCFederatedIdpInitLogoutTest.java [hunk 1]
```diff
@@ -0,0 +1,934 @@
...
+    public OIDCFederatedIdpInitLogoutTest(TestUserMode userMode) throws Exception {
+
+        context = new AutomationContext("IDENTITY", userMode);
+    }
+
+    @BeforeClass(alwaysRun = true)
+    public void initTest() throws Exception {
+
+        super.init();
+//        changeISConfiguration();
```
Comment (piraveena) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCFederatedIdpInitLogoutTest.java:183]: why it is commented? We need to start the server with tenant qualified URL enabled

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCFederatedIdpInitLogoutTest.java [hunk 2]
```diff
@@ -0,0 +1,934 @@
...
+        applicationManagementServiceClients = new HashMap<>();
+        oAuthAdminClients = new HashMap<>();
+        tenantServiceClient = new TenantManagementServiceClient(isServer.getContextUrls().getBackEndUrl(),
+                sessionCookie);
+        createTenants();
+        backendURL = context.getContextUrls().getBackEndUrl();
+        AuthenticatorClient logManger = new AuthenticatorClient(backendURL);
+        String primaryTenantCookie = logManger
+                .login("primaryAdmin@primary.com", "password", isServer.getInstance().getHosts().get("default"));
```
Comment (piraveena) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCFederatedIdpInitLogoutTest.java:192]: make this to a constant- PrimaryUser username

**Comment by piraveena** [comment]
please link the git issue here

File: modules/integration/tests-integration/tests-backend/src/test/resources/artifacts/IS/oidc/tenant_qualified_paths_enabled.toml [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+enable_tenanted_sessions = true
+
+# The KeyStore which is used for encrypting/decrypting internal data. By default the primary keystore is used as the internal keystore.
+
+#[keystore.internal]
+#file_name = "$ref{keystore.primary.file_name}"
+#type = "$ref{keystore.primary.type}"
+#password = "$ref{keystore.primary.password}"
+#alias = "$ref{keystore.primary.alias}"
+#key_password = "$ref{keystore.primary.key_password}"
```
Comment (piraveena) [inline modules/integration/tests-integration/tests-backend/src/test/resources/artifacts/IS/oidc/tenant_qualified_paths_enabled.toml:56]: remove these commented configs

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCFederatedIdpInitLogoutTest.java [hunk 4]
```diff
@@ -0,0 +1,998 @@
...
+    // Urls related to service provider configured in Primary IS.
+    private static final String PRIMARY_IS_SP_NAME = "playground2";
+    private static final String PRIMARY_IS_SP_AUTHENTICATION_TYPE = "federated";
+    private static final String PRIMARY_IS_SP_CALLBACK_URL =
+            "http://localhost:8490/playground2/oauth2-authorize-user.jsp";
+    private static final String PRIMARY_IS_SP_BACK_CHANNEL_LOGOUT_URL = "http://localhost:8490/playground2/bclogout";
+
+    // Urls related to service provider configured in Federated IS.
+    private static final String FEDERATED_IS_SP_NAME = "travelocity";
+    private static final String FEDERATED_IS_SP_CALLBACK_URL = "http://localhost:8490/travelocity.com/home.jsp";
```
Comment (piraveena) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCFederatedIdpInitLogoutTest.java:125]: why do we use travelocity? need to use playground callbackurl for oauth/oidc flows

Comment (ThaminduR) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCFederatedIdpInitLogoutTest.java]: changed the application names to **Application 1** and **Application 2**, and used `https://localhost/` as callbackUrl and back-channel logout url.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCFederatedIdpInitLogoutTest.java [hunk 5]
```diff
@@ -0,0 +1,1051 @@
...
+        // Get the id token from the primary is.
+        OIDCTokens tokens = getIdTokenFromIdp(authorizationCode, primSPClientID, primSPClientSecret,
+                PRIMARY_IS_SP_CALLBACK_URL, PRIMARY_IS_TOKEN_ENDPOINT);
+        primaryIdToken = tokens.getIDTokenString();
+        String primaryAccessToken = tokens.getAccessToken().getValue();
+        Assert.assertNotNull(primaryIdToken, "Primary sp Id token is null.");
+        // Extract claims from id token.
+        SignedJWT signedJWT = SignedJWT.parse(primaryIdToken);
+        JWTClaimsSet jwtClaimsSet = signedJWT.getJWTClaimsSet();
+        primaryIsk = (String) jwtClaimsSet.getClaim("isk");
```
Comment (piraveena) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCFederatedIdpInitLogoutTest.java:289]: move isk to a constant

**Pull request closed without merging**